pkg/cluster/nodes/node.go

```diff
@@ -403,42 +403,53 @@ func (n *Node) WriteFile(dest, content string) error {
 	return nil
 }
 
-var proxyEnvs = []string{"HTTP_PROXY", "HTTPS_PROXY", "NO_PROXY"}
+// NeedProxy returns true if the host environment appears to have proxy settings
+func NeedProxy() bool {
+	details := getProxyDetails()
+	return len(details) > 0
+}
 
 // SetProxy configures proxy settings for the node
 //
 // Currently it only creates systemd drop-in for Docker daemon
 // as described in Docker documentation: https://docs.docker.com/config/daemon/systemd/#http-proxy
 //
-// See also: NeedProxy
+// See also: NeedProxy and getProxyDetails
 func (n *Node) SetProxy() error {
+	details := getProxyDetails()
 	// configure Docker daemon to use proxy
 	proxies := ""
-	for _, name := range proxyEnvs {
-		val := os.Getenv(name)
-		if val != "" {
-			proxies += fmt.Sprintf("\"%s=%s\" ", name, val)
-		}
+	for key, val := range details {
+		proxies += fmt.Sprintf("\"%s=%s\" ", key, val)
 	}
 
-	if proxies != "" {
-		err := n.WriteFile("/etc/systemd/system/docker.service.d/http-proxy.conf",
-			"[Service]\nEnvironment="+proxies)
-		if err != nil {
-			errors.Wrap(err, "failed to create http-proxy drop-in")
-		}
+	err := n.WriteFile("/etc/systemd/system/docker.service.d/http-proxy.conf",
+		"[Service]\nEnvironment="+proxies)
+	if err != nil {
+		errors.Wrap(err, "failed to create http-proxy drop-in")
 	}
 
 	return nil
 }
 
-// NeedProxy returns true if the host environment appears to have proxy settings
+// getProxyDetails returns a struct with the host environment proxy settings
```

Comment:
s/struct with/map containing/, this is not a struct :upside_down_face: 
```suggestion
// getProxyDetails returns a map containing the host environment proxy settings
```
it's also worth considering making this _actually_ return a struct instead and ignoring the comment about renaming above then

```go
// proxyDetails contains proxy settings discovered on the host
type proxyDetails struct {
    Envs map[string]string
    // future proxy details here
}

func getProxyDetails() proxyDetails {
```